examples/rosettacode/ethiopian_multiplication.nit

```diff
@@ -0,0 +1,51 @@
+#!/usr/bin/env nit
+#
+# This file is part of NIT ( http://www.nitlanguage.org ).
+# This program is public domain
+
+# Task: Ethiopian multiplication
+# SEE: <http://rosettacode.org/wiki/Ethiopian_Multiplication>
+
+module ethiopian_multiplication
+
+redef class Int
+
+	fun half: Int
+	do
+		return self/2
+	end
+
+	fun double: Int
+	do
+		return self*2
+	end
+
+	fun isEven: Bool
+	do
+		return self%2 == 0
+	end
+end
+
+fun ethiopian(x, y: Int): Int
+   	do
+    	var t = new HashMap[Int, Int]
+    	t[x] = y
```

Comment:
Done. :)

- replaced isEven to is_even
- print t

On Thu, Jun 11, 2015 at 8:56 AM, Jean Privat <notifications@github.com>
wrote:

> In examples/rosettacode/ethiopian_multiplication.nit
> <https://github.com/privat/nit/pull/1464#discussion_r32216980>:
>
> > +
> > +	fun double: Int
> > +	do
> > +		return self*2
> > +	end
> > +
> > +	fun isEven: Bool
> > +	do
> > +		return self%2 == 0
> > +	end
> > +end
> > +
> > +fun ethiopian(x, y: Int): Int
> > +   	do
> > +    	var t = new HashMap[Int, Int]
> > +    	t[x] = y
>
> I do not understand what t is used for.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/privat/nit/pull/1464/files#r32216980>.
>